### ISSUE TITLE
Using setDimensions on Canvas with isDrawingMode leaves a ghost path

### ISSUE THREAD

**Issue by technotablet** [fabricjs/fabric.js#5890]
## Version
3.4.0

## Test Case
Working OK in 3.2.0 - https://jsfiddle.net/marquivon/s0zL28x9/2/
Working NOT OK 3.4.0 - https://jsfiddle.net/marquivon/b342mg6h/

## Information about environment
Nodejs or browser? **Browser**
Which browsers? **Firefox, Safari, Chrome**

## Steps to reproduce
1. Create a Canvas with the following options
```
var canvas = new fabric.Canvas("c");
canvas.isDrawingMode = true;
// Brush size WITH opacity - important to view difference
canvas.freeDrawingBrush.color = "rgba(255, 0, 0, 0.5)";
canvas.freeDrawingBrush.width = 5;
```
2. Draw something on the canvas
3. Resize the canvas using the following
```
canvas.setDimensions({
  width: 450,
  height: 450
});
canvas.requestRenderAll();
```
4. Draw again. On mouse up, you'll see a **darker** brush on top of the original path.

## Expected Behavior
There should not be a ghost path on top of the original path.
Once the path is created, on mouse up, no additional ghost path is visible.

In version 3.2.0 it is working as expected - Example: https://jsfiddle.net/marquivon/s0zL28x9/2/

## Actual Behavior
* There is a ghost path that is created.
* The ghost path is not available as an object when you do `canvas.getObjects()`, so it is not yet "registered" as an object.
* It is removed as soon as another path is created, or any selection action is performed on the canvas.

**Comment by technotablet** [comment]
Is there a workaround where I can refresh the canvas after mouse up? The statement
`canvas.requestRenderAll()`
doesn't seem to help. Thank you for any suggestions.

**Comment by asturur** [comment]
ok definitely a bug, a weird bug.

**Comment by alexkonst** [comment]
I think PR #5866 fix this issue.

**Comment by technotablet** [comment]
Thanks, @alexkonst. I just tried it out, and this minor change made a significant difference! Thank you! I guess as @asturur mentioned in the PR, once he's is confident about this change, then we might see it in an upcoming release.

**Comment by asturur** [comment]
sorry for letting this behind, i ll be back to it as soon as possible

**Comment by asturur** [comment]
fixed with #5866
